### PR TITLE
os/mm/mm_heap/mm_getheap.c : Fix string copy size

### PR DESCRIPTION
**Pull request by jeongchanKim**
For security issue, it is better to copy the string based on its size.

### CODE REVIEW

**Review by jeongarmy — DISMISSED** [review]
LGTM

File: os/mm/mm_heap/mm_getheap.c [hunk 1]
```diff
@@ -84,7 +84,7 @@ void mm_add_app_heap_list(struct mm_heap_s *heap, char *app_name)
 
 	node->is_active = true;
 	node->heap = heap;
-	strncpy(node->app_name, app_name, BIN_NAME_MAX);
+	strncpy(node->app_name, app_name, strlen(app_name) + 1);
```
Comment (sunghan-chang) [inline os/mm/mm_heap/mm_getheap.c:87]: I don't know why this change is necessary. I think previous is right.

File: os/mm/mm_heap/mm_getheap.c [hunk 2]
```diff
@@ -144,7 +145,7 @@ struct mm_heap_s *mm_get_app_heap_with_name(char *app_name)
 
 	/* Search the heap */
 	while (node && node->is_active) {
-		if (strncmp(node->app_name, app_name, BIN_NAME_MAX) == 0) {
+		if (strncmp(node->app_name, app_name, BIN_NAME_MAX + 1) == 0) {
```
Comment (Taejun-Kwon) [inline os/mm/mm_heap/mm_getheap.c:148]: last is always null, BIN_NAME_MAX is correct, I think.

Comment (jeongchanKim) [reply os/mm/mm_heap/mm_getheap.c:148]: Let's assume that `node->app_name` is "abc", `app_name` is "abcd" and BIN_NAME_MAX is 3.
Then this comparison returns OK, because it tries to check from index 0 to 2.
But it is not correct. 
So I think `BIN_NAME_MAX + 1` is correct.

Comment (Taejun-Kwon) [reply os/mm/mm_heap/mm_getheap.c:148]: OK

**Review by an4967 — DISMISSED** [review]
LGTM

**Review by Taejun-Kwon — DISMISSED** [review]

File: os/mm/mm_heap/mm_getheap.c [hunk 3]
```diff
@@ -43,7 +43,7 @@
 typedef struct app_heap_s {
 	struct app_heap_s *flink;
 	struct app_heap_s *blink;
-	char app_name[BIN_NAME_MAX];
```
Comment (sunghan-chang) [inline os/mm/mm_heap/mm_getheap.c:46]: There are many locations which use BIN_NAME_MAX for array size. But you modify only here. If this is necessary, you should check all of others. But is this really necessary?

**Review by jsdosa — APPROVED** [review]
Looks nice to me.

**Review by jsdosa — APPROVED** [review]
Looks nice to me
